backend/src/campaigns.rs

```diff
@@ -157,3 +158,44 @@ pub async fn roles(
 
     Ok(Json(RolesResponse { roles }))
 }
+
+#[post("/<campaign_id>/roles", data = "<role>")]
+pub async fn create_role(
+    campaign_id: i32,
+    mut role: Form<CreateRoleInput>,
+    user: User,
+    db: Database,
+) -> Result<(), Json<RolesError>> {
+    let campaign = db
+        .run(move |conn| Campaign::get_from_id(conn, campaign_id))
+        .await;
+
+    let campaign = campaign.ok_or(Json(RolesError::CampaignNotFound))?;
```

Comment:
don't need to do .ok_or on new line. Can chain it with await:

.run()
.await
.ok_or()?;

Reply:
Yeah, I actually copied this style from the get function of the same route - generally do chain ok_or for this type of thing. Will change.